src/components/sharing/util.js

```diff
@@ -85,8 +96,17 @@ export const getSharingFns = (type) => {
                 getPermList: (sharing) => sharing["user-permissions"],
                 getUserId: (permission) => permission.user,
                 formatSubject: (user) => {
+                    let subject = {
+                        source_id: "ldap",
+                        id: user.id,
+                    };
+                    if (user.source_id) {
+                        subject.source_id = user.source_id;
+                    } else {
+                        subject = irodsNameToSubject(user.id);
+                    }
                     return {
-                        user: user.id,
+                        subject: subject,
```

Comment:
FYI, this is some equivalent syntactic sugar:
```suggestion
                        subject,
```

Reply:
Good point. I think I was constructing it down here before and eventually rearranged it how this is, but didn't update it for the cleaner syntax.